core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -768,6 +768,12 @@ private[chisel3] object Builder extends LazyLogging {
       for ((elt, i) <- iter.zipWithIndex) {
         nameRecursively(s"${prefix}_${i}", elt, namer)
       }
+    case product: Product =>
+      product.productIterator.zipWithIndex.foreach {
```

Comment:
Rather than using `.zipWithIndex`, I think it would be safer to `.zip(product.productElementNames)`, then this will do sensible things for non-Tuple products (eg. every case class)